### DISCUSSION TITLE
Should zed support swipe between pages?

### DISCUSSION THREAD

**Discussion by VladasZ** [zed-industries/zed#12531] [Feature Requests]
macOS has a feature allowing swipes between pages in application:
<img width="483" alt="image" src="https://github.com/zed-industries/zed/assets/10122409/ed8bab01-40d9-4209-b292-378b36a3864e">

I think it would be cool and convenient if `zed` could trigger `pane::GoBack` and `pane::GoForward` event on such swipes.

But I don't know what is the best way to implement it. I googled that `Cocoa` doesn't provide clear event for 2 finger left and right gestures:
https://stackoverflow.com/questions/6747603/how-to-implement-the-two-finger-swipe-gesture-in-cocoa-to-go-back-and-forward

We can detect 3 finger swipes or detect high `delta.x` values of `ScrollWheelEvent`.
But this can be annoying if this event will interfere with horizontal scrolling. This can be mitigated by setting some threshold for `delta.x`.

This feature can be disabled by default by some option to not confuse current users. 

Also I don't know what is the best place to add such listener in the application.

**Comment by seivan** [comment]
The answer is, yes, it should. Everything else on macOS does it. Even third party applications like Discord which is on Electron, does it. 

Also, the three finger option is the only one that actually works as swiping, the others behave... differently, try it out on Xcode. 
You don't need to bother with `ScrollWheelEvent`, it has its own swipe-event. 

Finally finished up the [old pr](https://github.com/zed-industries/zed/pull/14939) and made a new https://github.com/zed-industries/zed/pull/36574

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by seivan** [comment]
It’s funny. I added support but it wasn’t accepted because there wasn’t an example crate on how to use it with other apps using the gpu crate . But Zed was the example, lol
